onnxruntime/contrib_ops/rocm/math/binary_elementwise_ops.cc

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "binary_elementwise_ops.h"
+#include "binary_elementwise_ops_impl.h"
+
+using namespace onnxruntime::common;
+namespace onnxruntime {
+namespace contrib {
+namespace rocm {
+
+#define CONTRIB_BINARY_ELEMENTWISE_REGISTER_KERNEL_TYPED(x, ver, T)             \
+  ONNX_OPERATOR_TYPED_KERNEL_EX(                                                \
+      x,                                                                        \
+      kMSDomain,                                                                \
+      ver,                                                                      \
+      T,                                                                        \
+      kRocmExecutionProvider,                                                   \
+      KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<T>()), \
+      x<T>);
+
+#define CONTRIB_BINARY_ELEMENTWISE_COMPUTE(x, T)                                                                         \
+  template <>                                                                                                    \
+  Status x<T>::ComputeInternal(OpKernelContext* context) const {                                                 \
+    BinaryElementwisePreparation prepare(this);                                                                  \
+    Prepare(context,  &prepare);                                                          \
+    ORT_RETURN_IF_ERROR(prepare.CopyToGpu());                                                                    \
```

Comment:
It's not present for CUDA case?

Reply:
At present, the implementation of binary elementwise kernel is different on CUDA vs ROCm due to perf issue caused by ROCm compiler. AMD plans to release the fix in ROCm 3.10.